lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,486 @@
+        except vim.fault.InvalidLogin as e:
+            raise AnsibleError("Connection Login Error: %s" % to_native(e.msg))
+
+    def _establish_vm(self):
+        searchIndex = self._si.content.searchIndex
+        self.vm = searchIndex.FindByInventoryPath(self.get_option("vm_path"))
+
+        if self.vm is None:
+            raise AnsibleError("Unable to find VM by path '%s'" % to_native(self.get_option("vm_path")))
+
+        self.vm_auth = vim.NamePasswordAuthentication(
+            username=self.get_option("vm_user"), password=self.get_option("vm_password"), interactiveSession=False
+        )
+
+        try:
+            self.authManager.ValidateCredentialsInGuest(vm=self.vm, auth=self.vm_auth)
+        except vim.fault.InvalidPowerState as e:
+            raise AnsibleError("VM Power State Error: %s" % to_native(e.msg))
+        except vim.fault.RestrictedVersion as e:
+            raise AnsibleError("Restricted Version Error: %s" % to_native(e.msg))
+        except vim.fault.GuestOperationsUnavailable as e:
+            raise AnsibleError("VM Guest Operations (VMware Tools) Error: %s" % to_native(e.msg))
+        except vim.fault.InvalidGuestLogin as e:
+            raise AnsibleError("VM Login Error: %s" % to_native(e.msg))
+        except vim.fault.NoPermission as e:
+            raise AnsibleError("No Permission Error: %s %s" % (to_native(e.msg), to_native(e.privilegeId)))
+
+    def _connect(self):
+        if not HAS_REQUESTS:
+            raise AnsibleError("%s : %s" % (missing_required_lib('requests'), REQUESTS_IMP_ERR))
+
+        if not HAS_PYVMOMI:
+            raise AnsibleError("%s : %s" % (missing_required_lib('PyVmomi'), PYVMOMI_IMP_ERR))
+
+        super(Connection, self)._connect()
+
+        if self.connected:
+            pass
+
+        self._establish_connection()
+        self._establish_vm()
+
+        self._connected = True
+
+    def close(self):
+        """Close connection."""
+        super(Connection, self).close()
+
+        Disconnect(self._si)
+        self._connected = False
+
+    def reset(self):
+        """Reset the connection."""
+        super(Connection, self).reset()
+
+        self.close()
+        self._connect()
+
+    def create_temporary_file_in_guest(self, prefix="", suffix=""):
+        """Create a temporary file in the VM."""
+        try:
+            return self.fileManager.CreateTemporaryFileInGuest(vm=self.vm, auth=self.vm_auth, prefix=prefix, suffix=suffix)
+        except vim.fault.NoPermission as e:
+            raise AnsibleError("No Permission Error: %s %s" % (to_native(e.msg), to_native(e.privilegeId)))
+
+    def _get_program_spec_program_path_and_arguments(self, cmd):
+        if self.windowsGuest:
+            cmd_parts = self._shell._encode_script(cmd, as_list=False, strict_mode=False, preserve_rc=False)
+
+            program_path = "cmd.exe"
+            arguments = "/c %s" % cmd_parts
+        else:
+            program_path = self._play_context.executable
```

Comment:
dont use play_context, add 'executable' option to docs and `self.get_option('executable')`

Reply:
What should be the default value?

Reply:
I think @bcoca something like this - https://github.com/ansible/ansible/blob/166a33ef4d7272ed83c0b365545ccaccc50dd6d5/lib/ansible/plugins/connection/lxd.py#L23

Reply:
Done.